Source/OxyPlot/Rendering/RenderContext/RenderContextBase.cs

```diff
@@ -228,21 +230,42 @@ public virtual void CleanUp()
         {
         }
 
-        /// <summary>
-        /// Sets the clip rectangle.
-        /// </summary>
-        /// <param name="rect">The clip rectangle.</param>
-        /// <returns>True if the clip rectangle was set.</returns>
-        public virtual bool SetClip(OxyRect rect)
+        /// <inheritdoc/>
+        public virtual void PopClip()
         {
-            return false;
+            if (this.clipStack.Count == 0)
+            {
+                throw new InvalidOperationException($"Unbalanced call to {nameof(PopClip)}.");
+            }
+
+            this.ResetClip();
+            this.clipStack.Pop();
+            if (this.clipStack.Count > 0)
+            {
+                this.SetClip(this.clipStack.Peek());
```

Comment:
Yes I think you are right.

Also I think it the implementation in the XKCD decorator should override `PushClip` and `PopClip` directly, rather than `SetClip` and `ResetClip`.